### PR TITLE
Rename annotation to `gateway.kusk.io`

### PR DESCRIPTION
**Pull request by mbana**
Use public Kubernetes annotations by having keys, e.g., (`gateway.kusk.io`), that contain periods.

Partially closes kubeshop/kusk#49.

See <https://github.com/kubeshop/kusk/issues/49>

Signed-off-by: Mohamed Bana <mohamed@bana.io>

---

**Note:** Please do not merge until the corresponding change has been made in <https://github.com/kubeshop/kusk-gateway/blob/main/internal/controllers/service_controller.go#L62>.

---

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by aabedraba — APPROVED** [review]
👍🏽

**Review by nicufk — APPROVED** [review]

**Comment by kylehodgetts** [comment]
Please don't merge until the tests are passing and the PR is merged here https://github.com/kubeshop/kusk-gateway/pull/528

**Pull request closed without merging**
